Sources/RoutingKit/Routing/TrieRouter+Node.swift

```diff
@@ -7,7 +7,7 @@ extension TrieRouter {
         var value: String
```

Comment:
I don't think we need to store this value anymore since it's stored in the dictionary.

Reply:
We should also rename `TrieRouter` now that it is no longer a trie router haha. Maybe just `Router`?

Reply:
Isn't it still a trie? It's just that any node that held constants now holds them in a dictionary instead of an array.

Reply:
ah yeah, good point.

Reply:
I think the comment regarding storing the value in the node is still valid though.

Reply:
I think it's worth keeping around to generate Node descriptions. I could get rid of value if you really want it gone and refactor how we're generating descriptions though if you like.

Reply:
I think that would be smart. By storing the string twice, we're effectively doubling the physical size of the tree which could have performance implications in itself. I can also take a look at refactoring the descriptions if you want. That method was mind boggling to write haha.

Reply:
Alright sounds good. I'll take care of it later tonight.

Reply:
I took another look and I really don't think it's feasible.
```swift
if let parameter = currentNode.parameter {
    // if no constant routes were found that match the path, but
    // a dynamic parameter child was found, we can use it
    let value = ParameterValue(
        slug: parameter.value, // THIS HERE: Not a constant, so it's not in the dictionary
        value: path
    )
    parameters.values.append(value)
    currentNode = parameter
    continue search
}
```

Our description generator would have to be entirely within the TrieRouter rather than being able to lean on the fact that nodes can recursively generate their own descriptions. We'd also have to hold node values as we traversed the trie. We can revisit this later, but it would require major restructuring for not that much gain at this point. 

Maybe not as important, but holding Node values outside the respective nodes themselves feels like bad modeling to me. It makes things kind of confusing to reason about.

Reply:
@twof check out my impl here and lmk what you think: https://github.com/vapor/routing/pull/62

Reply:
#62 has been merged. Marking this as resolved.